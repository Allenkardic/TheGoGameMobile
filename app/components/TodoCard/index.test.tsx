// import React from 'react';
// import renderer from 'react-test-renderer';
// import TodoCard from '.';
// import {Text} from '../Typography';
// import Icon from 'react-native-vector-icons/Feather';

// describe('TodoCard Component', () => {
//   const mockEditPress = jest.fn();
//   const mockDeletePress = jest.fn();

//   it('renders correctly with all props', () => {
//     const component = renderer.create(
//       <TodoCard
//         title="Sample Todo"
//         body="This is a sample todo item."
//         date={new Date().toString()}
//         onPressEdit={mockEditPress}
//         onPressDelete={mockDeletePress}
//       />,
//     );
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot(); // Snapshot test to ensure no unexpected changes
//   });

//   it('calls onPressEdit when edit icon is pressed', () => {
//     const component = renderer.create(
//       <TodoCard
//         title="Sample Todo"
//         body="This is a sample todo item."
//         date={new Date().toString()}
//         onPressEdit={mockEditPress}
//         onPressDelete={mockDeletePress}
//       />,
//     );
//     const root = component.root;

//     // Simulate Edit icon press
//     const editIcon = root.findAllByType(Icon)[0];
//     editIcon.props.onPress();
//     expect(mockEditPress).toHaveBeenCalledTimes(1); // Ensure the onPressEdit was called
//   });

//   it('calls onPressDelete when delete icon is pressed', () => {
//     const component = renderer.create(
//       <TodoCard
//         title="Sample Todo"
//         body="This is a sample todo item."
//         date={new Date().toString()}
//         onPressEdit={mockEditPress}
//         onPressDelete={mockDeletePress}
//       />,
//     );
//     const root = component.root;

//     //  Simulate delete icon press
//     const deleteIcon = root.findAllByType(Icon)[1];
//     deleteIcon.props.onPress();
//     expect(mockDeletePress).toHaveBeenCalledTimes(1); // Ensure the onPressDelete was called
//   });

//   it('formats the date correctly', () => {
//     const date = new Date().toString();
//     const component = renderer.create(
//       <TodoCard
//         title="Sample Todo"
//         body="This is a sample todo item."
//         date={date}
//         onPressEdit={mockEditPress}
//         onPressDelete={mockDeletePress}
//       />,
//     );
//     const root = component.root;

//     // Check if the date is displayed in the correct format
//     const dateText = root
//       .findAllByType(Text)
//       .find((text: any) =>
//         text.props.children.includes(new Date(date).toDateString()),
//       );
//     expect(dateText).not.toBeUndefined();
//     if (dateText) {
//       expect(dateText.props.children).toBe(new Date(date).toDateString());
//     }
//   });
// });
