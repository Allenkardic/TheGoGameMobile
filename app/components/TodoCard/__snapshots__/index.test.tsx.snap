// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard Component renders correctly with all props 1`] = `
<View
  style={
    {
      "borderColor": "#BDBDBD",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 1,
      "marginBottom": 24,
      "marginHorizontal": 24,
      "paddingHorizontal": 12,
      "paddingVertical": 24,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-Regular",
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#010a1a",
        },
        undefined,
        {
          "fontFamily": "Inter-Medium",
        },
        {
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Sample Todo
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-Regular",
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#36454f",
        },
        {
          "marginBottom": 12,
          "marginTop": 24,
        },
        null,
        {
          "fontSize": 14,
          "lineHeight": 21,
        },
      ]
    }
  >
    This is a sample todo item.
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "color": "#6D6E71",
          },
          undefined,
          null,
          {
            "fontSize": 14,
            "lineHeight": 21,
          },
        ]
      }
    >
      Fri Dec 06 2024
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[MockFunction]}
        selectable={false}
        style={
          [
            {
              "color": "#139E5C",
              "fontSize": 20,
            },
            {
              "alignSelf": "flex-end",
            },
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        onPress={[MockFunction]}
        selectable={false}
        style={
          [
            {
              "color": "#E80808",
              "fontSize": 20,
            },
            [
              {
                "alignSelf": "flex-end",
              },
              {
                "marginLeft": 24,
              },
            ],
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
